addons/html_editor/static/src/others/embedded_components/core/table_of_content/table_of_content.xml

```diff
@@ -1,16 +1,24 @@
 <?xml version="1.0" encoding="utf-8"?>
 <templates id="template" xml:space="preserve">
 <t t-name="html_editor.EmbeddedTableOfContent">
-    <div class="o_embedded_toc_content">
-        <t t-foreach="this.state.toc.headings" t-as="heading" t-key="heading_index">
-            <a href="#" contenteditable="false"
-                t-out="heading.name"
-                t-on-click.prevent="() => this.onTocLinkClick(heading)"
-                t-attf-class="o_no_link_popover py-1 pe-1 d-block o_embedded_toc_link #{'o_embedded_toc_link_depth_' + heading.depth}"/>
-        </t>
+    <div class="px-1" t-att-class="{ 'pb-1' : !state.folded and state.toc.headings.length }">
```

Comment:
let's alter the condition like this:
```suggestion
    <div class="px-1" t-att-class="{ 'pb-1' : !state.folded and !displayTocHint() }">
```

so that it also work in readonly, good catch!

Reply:
It will lead to unnecessary padding when the readonly toc with no headings is clicked. I suppose the earlier condition ```!state.folded and state.toc.headings.length``` is good as the hint now is in a separate div, what do you think?

Folded: 
![image](https://github.com/user-attachments/assets/359bd2ef-dcb3-4353-a1ed-8ed3b0075fe1)
Unfolded:
![image](https://github.com/user-attachments/assets/b12f42dd-9852-4faa-ab4d-63f0a609762c)

Reply:
shouldnt the condition be `!state.folded or !displayTocHint()` by the way? we only want the padding if it is unfolded or if the hint is displayed

Reply:
For the readonly toc, it would add a padding when unfolded and remove the padding when unfolded. Problem will be when headings.length is 0. This way the toc has no headings but unfolding it will add an extra padding :)

We want add padding only when there are headings and when the toc is unfolded. And for the padding below the hint, we have the separate div now.